pkg/api/persistentvolumeclaim/util.go

```diff
@@ -151,3 +154,23 @@ func allocatedResourcesInUse(oldPVC *core.PersistentVolumeClaim) bool {
 
 	return false
 }
+
+func GetWarningsForPersistentVolumeClaim(pv *core.PersistentVolumeClaim) []string {
+	if pv == nil {
+		return nil
+	}
+
+	return GetWarningsForPersistentVolumeClaimSpec(field.NewPath("spec"), pv.Spec)
+}
+
+func GetWarningsForPersistentVolumeClaimSpec(fieldPath *field.Path, pvSpec core.PersistentVolumeClaimSpec) []string {
+	storageValue := pvSpec.Resources.Requests[core.ResourceStorage]
```

Comment:
do we also need to look at `pvSpec.Resources.Limits`?

Reply:
Added.

https://github.com/kubernetes/kubernetes/blob/9c4905e7af0d070d139c06638ab9868fd3521883/plugin/pkg/admission/limitranger/admission.go#L484
At first, I saw this in comments and not add the limit checking.